### PR TITLE
List blog posts

### PR DESCRIPTION
**Pull request by dthisner**
Adding functionality to list blog posts.

It grabs blog post from Bloggers API and lists them

Added Unit Tests

### CODE REVIEW

**Comment by thm-design** [comment]
Hi hi

File: src/api/blogger.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import Axios from 'axios'
+
+const baseUrl = 'https://www.googleapis.com/blogger/v3/blogs'
```
Comment (thm-design) [inline src/api/blogger.js:3]: Nit: this could also be stored in an env var.

File: src/components/App.js [hunk 2]
```diff
@@ -11,7 +11,7 @@ class App extends Component {
       <BrowserRouter>
         <Header />
         <Router />
-        <Footer />
+        {/* <Footer /> */}
```
Comment (thm-design) [inline src/components/App.js:14]: How dare you.

File: src/pages/BlogPosts.js [hunk 3]
```diff
@@ -1,11 +1,137 @@
...
+  const [pagnation, setPagnation] = useState({next: null, prev: null})
+
+  useEffect(() => {
+    const getData = async () => {
+      const response = await getPosts()
+
+      updatePostData(response)
+    }
+    getData()
+  }, [])
```
Comment (thm-design) [inline src/pages/BlogPosts.js:19]: There is no error handling here, and no cancellation of the network request in the event this component is un-mounted.
Here is an example with error handling and cancellation: 

`const [apiData, setApiData] = useState(null);

    useEffect(() => {
        const source = axios.CancelToken.source();

        axios.get("https://jsonplaceholder.typicode.com/todos", {
            cancelToken: source.token
        }).then(response => {
            setApiData(response.data);
        }).catch(err => {
            console.log("Catched error: " + e.message);
        });

        return () => {
            source.cancel("Component got unmounted");
        };
    }, []);`

Source: https://since1979.dev/cancel-axios-request-to-prevent-react-from-yelling-at-you

File: src/pages/BlogPosts.js [hunk 4]
```diff
@@ -1,11 +1,137 @@
...
+    }
+
+    return <div align="center">{renderData}</div>
+  }
+
   return (
     <div>
       <h1 role="heading">Blog Posts</h1>
+      <RenderBlogposts blogPosts={blogPosts} />
+      <div>{RenderPagnation(pagnation)}</div>
```
Comment (thm-design) [inline src/pages/BlogPosts.js:70]: Nit: So react components are usually named with uppercase camel,` <MyThing />` etc..  While functions are usually small case camel, `myThing()`.

Here you are using the component casing convention and using the function style ('RenderPagnation(pagnation)'), which works, but in the light of consistency and less confusion you could just do this: `<div><RenderPagnation pagnation={pagnation} />}</div>`

But don't forget to add curly brackets around the "prop" if you will use it as a component: `const RenderPagnation = ({ pagnation }) => {` etc.

**Review by thm-design — CHANGES_REQUESTED** [review]

File: src/pages/BlogPosts.js (hunk 4 above)
Comment (thm-design) [reply src/pages/BlogPosts.js:70]: PS: I know you already know this above ^

File: src/components/App.js (hunk 2 above)
Comment (dthisner) [reply src/components/App.js:14]: it asked for it! :D

**Pull request merged**